### PR TITLE
cpprestsdk: honor shared=False

### PR DESCRIPTION
**Pull request by SpaceIm**
see https://github.com/conan-io/conan-center-index/pull/13962#discussion_r1024410687

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/cpprestsdk/all/conandata.yml [hunk 1]
```diff
@@ -1,40 +1,40 @@
 sources:
-  "2.10.15":
```
Comment (garethsb) [inline recipes/cpprestsdk/all/conandata.yml:2]: I'm sure I've seen the guidance that _conandata.yml_ amd _config.yml_ should be reordered so newest versions are listed first, but right now I can't find it, e.g., in https://github.com/conan-io/conan-center-index/blob/edacf97622e46921af5f1717776adfeb0a129135/docs/adding_packages/conandata_yml_format.md 😕

**Review by garethsb — APPROVED** [review]
LGTM

File: recipes/cpprestsdk/all/conandata.yml (hunk 1 above)
Comment (prince-chrismc) [reply recipes/cpprestsdk/all/conandata.yml:2]: It's in the package template right now

**Review by prince-chrismc — APPROVED** [review]

File: recipes/cpprestsdk/all/conandata.yml (hunk 1 above)
Comment (garethsb) [reply recipes/cpprestsdk/all/conandata.yml:2]: Ah, e.g. here:
https://github.com/conan-io/conan-center-index/blob/fc7fdca06847ae16a865d6dc541738fff38a5d59/docs/package_templates/cmake_package/all/conandata.yml#L2

**Review by uilianries — APPROVED** [review]
LGTM

**Pull request merged**